### PR TITLE
client: Add README

### PR DESCRIPTION
**Pull request by zos**
Add README file to ir-client crate.

### CODE REVIEW

File: ir-client/README.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+implements part of
+[OCI Distribution](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#endpoints)
+client API. This allows to separately fetch manifest and blob files
+and list tags from the registry. This is not the main purpose of this
+library, thus it will not be described here.
+
+
+# Detailed description
+
+The main client provides below API:
```
Comment (Havner) [inline ir-client/README.md]: provides the following API

File: ir-client/README.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+## Client initialization
+
+```
```
Comment (Havner) [inline ir-client/README.md]: Try to find some tag for the code block to enable syntax highlighting for rust. Not sure it's possible, but it should be.

File: ir-client/README.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+host address and other tls related settings.
+
+## Image info fetching
+
+```
+    pub async fn get_image_info(
+        &self,
+        app_name: &str,
+        reference: Reference,
+    ) -> Result<ImageInfo, Error>\
```
Comment (Havner) [inline ir-client/README.md]: trailing backslash

File: ir-client/README.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+signature, version, license, etc).
+
+## Image files download
+
+```
+    pub async fn unpack_image(
+        &self,
+        image_info: &ImageInfo,
+        dest: impl AsRef<Path>,
+        temp: impl AsRef<Path>,
```
Comment (Havner) [inline ir-client/README.md:71]: something got cut here

File: ir-client/README.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        load_root_cert_store();
+    // Create token resolver for client certificate (impl ratls::InternalTokenResolver)
+    let token_resolver = Arc::new(get_token_resolver());
+    // Create ra-tls certificate resolver (impl rustls::ResolvesClientCert)
+    let cert_resolver = Arc::new(RaTlsCertResolver::from_token_resolver(token_resolver));
+
+    // Create client configuration
+    let config = Config::builder()
+        .host(conn.host)
+	    // Use ra-tls protocol for connection with server
```
Comment (Havner) [inline ir-client/README.md]: misaligned comment

**Review by Havner — CHANGES_REQUESTED** [review]

File: ir-client/README.md (hunk 1 above)
Comment (zos) [reply ir-client/README.md]: Done

File: ir-client/README.md (hunk 2 above)
Comment (zos) [reply ir-client/README.md]: Done

File: ir-client/README.md (hunk 3 above)
Comment (zos) [reply ir-client/README.md]: Done

File: ir-client/README.md (hunk 4 above)
Comment (zos) [reply ir-client/README.md:71]: Done

**Pull request merged**